File: Bubble sort/bub_sort.R
bubble_sort <- function(arr) {
  n <- length(arr)
  
  for (i in 1:(n-1)) {
    for (j in 1:(n-i)) {
      if (arr[j] > arr[j+1]) {
        # Swap arr[j] and arr[j+1]
        temp <- arr[j]
        arr[j] <- arr[j+1]
        arr[j+1] <- temp
      }
    }
  }
  
  return(arr)
}

# Test the function
arr <- c(64, 34, 25, 12, 22, 11, 90)
sorted_arr <- bubble_sort(arr)
print(paste("Sorted array is:", paste(sorted_arr, collapse = ", ")))

File: Bubble sort/bub_sort.py
def bubble_sort(arr):
    n = len(arr)
    for k in range(n):
        for ptr in range(0, n - k - 1):
            # Swap if the element found is greater
            if arr[ptr] > arr[ptr+1]:
                arr[ptr], arr[ptr+1] = arr[ptr+1], arr[ptr]
    return arr


if __name__ == "__main__":
   my_data = [64, 34, 25, 12, 22, 90, 1]
   print("Before sorting:", my_data)
   sorted_data = bubble_sort(my_data)
   print("After sorting:", sorted_data)
    
        